### PR TITLE
Fix Issue #717 Cannot set DefaultAuthorizer and have CORS enabled

### PR DESCRIPTION
**Pull request by xrendan**
*Issue #717*

*Description of changes:*
Add `AddDefaultAuthorizerToCorsPreflight` parameter Auth object so that you can in API  fixed code from PR #828 so that it does not fail the test cases. 

*Description of how you validated changes:*
Generated test cases for default condition and with `AddDefaultAuthorizerToCorsPreflight` set to False, the output cloudformation json has security for the options method when it is True and has no security when it is set to False.

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [x] Update documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by xrendan** [comment]
The tests failed due to API gateway hashes and I don't know how to fix that and I don't know how to generate test cases for aw-us-gov and aws-cn so I need direction or help on that.

**Comment by praneetap** [comment]
> The tests failed due to API gateway hashes and I don't know how to fix that and I don't know how to generate test cases for aw-us-gov and aws-cn so I need direction or help on that.

To fix the hashes could you try running the tests in python 2.7 and copying the output into the output/json, output/aws-cn/json and output/aws-us-gov/json files.
Generating test cases for aws-cn and aws-us-gov should happen automatically for your tests. You will see the output for those json files in the error output and would just need to copy over to the right path.

**Comment by praneetap** [comment]
@xrendan I fixed the tests and created a [pr](https://github.com/xrendan/serverless-application-model/pull/1) against your branch. Can you please merge it in? I don't have permissions to do so.

**Comment by xrendan** [comment]
I merged the PR for you, is there anything else that I need to do?

File: samtranslator/swagger/swagger.py [hunk 1]
```diff
@@ -408,14 +409,18 @@ def set_path_default_authorizer(self, path, default_authorizer, authorizers):
...
 
         for method_name, method in self.get_path(path).items():
+            normalized_method_name = self._normalize_method_name(method_name)
             # Excluding paramters section
-            if method_name == "parameters":
+            if normalized_method_name == "parameters":
                 continue
-            self.set_method_authorizer(path, method_name, default_authorizer, authorizers,
-                                       default_authorizer=default_authorizer, is_default=True)
+            if not (add_default_auth_to_preflight is False and normalized_method_name == "options"):
```
Comment (keetonian) [inline samtranslator/swagger/swagger.py]: This is a little confusing. Isn't the following functionally equivalent?

`if add_default_auth_to_preflight or normalized_method_name != "options"`:

File: tests/translator/test_translator.py [hunk 2]
```diff
@@ -265,57 +267,57 @@ class TestTranslatorEndToEnd(TestCase):
...
-        with patch('boto3.session.Session.region_name', region):
-            parameter_values = get_template_parameter_values()
-            mock_policy_loader = MagicMock()
-            mock_policy_loader.load.return_value = {
-                'AWSLambdaBasicExecutionRole': 'arn:{}:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole'.format(partition),
-                'AmazonDynamoDBFullAccess': 'arn:{}:iam::aws:policy/AmazonDynamoDBFullAccess'.format(partition),
-                'AmazonDynamoDBReadOnlyAccess': 'arn:{}:iam::aws:policy/AmazonDynamoDBReadOnlyAccess'.format(partition),
-                'AWSLambdaRole': 'arn:{}:iam::aws:policy/service-role/AWSLambdaRole'.format(partition),
-            }
+        with (patch('samtranslator.plugins.application.serverless_app_plugin.ServerlessAppPlugin._sar_service_call', mock_sar_service_call),
```
Comment (keetonian) [inline tests/translator/test_translator.py]: Thank you for this! I went ahead and fixed the merge conflicts that were happening in this file; could you make sure I did it right?

This should also fix the weird "unbound local variable" error that kept happening; I worked out a similar fix recently but you beat me to it 👍

**Comment by praneetap** [comment]
> I merged the PR for you, is there anything else that I need to do?

@xrendan  That would be it! Thanks. That should fix the build issues!

**Comment by keetonian** [comment]
Guess that didn't do it; I made the changes locally this time and submitted another PR to your branch: https://github.com/xrendan/serverless-application-model/pull/2

That should fix the test failures. Sorry about this!

**Comment by xrendan** [comment]
@keetonian it's all merged up, thanks for fixing my tests, I hadn't had the time to do it myself with work this week. I'll submit a PR for the unbound local issue later today or tomorrow.

File: samtranslator/swagger/swagger.py (hunk 1 above)
Comment (brettstack) [reply samtranslator/swagger/swagger.py]: Can we make this change

**Comment by leantorres73** [comment]
Maybe this is a silly question but I don't know the workflows. When is this change going to be available in AWS Cloudformation?

**Pull request merged**